### PR TITLE
Javalab backpack dropdown

### PR DESCRIPTION
**Pull request by JillianK**
Update the backpack button to use a backpack icon, create a dropdown menu with dummy filenames on button click with an import button.

Screen Shot 2021-05-26 at 2 43 11 PM
Screen Shot 2021-05-26 at 2 43 07 PM
Screen Shot 2021-05-26 at 2 49 09 PM

Update: light mode
Screen Shot 2021-05-27 at 1 30 24 PM

Design from Figma
Screen Shot 2021-05-26 at 2 52 31 PM

## Links
- spec: [figma](https://www.figma.com/file/fZo4fcZmDUioOCaz04Gza4/Javalab-Design-Spec?node-id=0%3A1)
- jira ticket: [Part of UI Pass](https://codedotorg.atlassian.net/browse/CSA-361?atlOrigin=eyJpIjoiNTljZTY2ODExMTA2NDYxNTlkMjU3ZjAyMWUyNDFlMDEiLCJwIjoiaiJ9)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/javalab/Backpack.jsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import color from '@cdo/apps/util/color';
+import onClickOutside from 'react-onclickoutside';
+
+const placeholderFiles = [
+  'MyClass.java',
+  'MyPainter.java',
+  'NeighborhoodMain.java'
+];
+
+const styles = {
```
Comment (molly-moen) [inline apps/src/javalab/Backpack.jsx]: I think we are moving towards putting styles at the bottom of files, let's start using that pattern for new files

File: apps/src/javalab/Backpack.jsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    borderRadius: 2,
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center'
+  },
+  buttonStyles: {
+    cursor: 'pointer',
+    float: 'right',
+    backgroundColor: color.light_purple,
+    marginTop: 3,
```
Comment (molly-moen) [inline apps/src/javalab/Backpack.jsx]: nit: these 4 margin styles can be combined into 1 like so: `margin: '3px 3px 3px 0px'` (order is top, right, bottom, left)

File: apps/src/javalab/Backpack.jsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      backgroundColor: color.cyan
+    }
+  },
+  dropdownOpenButton: {
+    backgroundColor: color.cyan
+  },
+  fileListItem: {
+    display: 'flex',
+    flexDirection: 'row',
+    paddingLeft: 10,
```
Comment (molly-moen) [inline apps/src/javalab/Backpack.jsx]: similarly we can combine padding styles here

File: apps/src/javalab/Backpack.jsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        >
+          <img
+            src="/blockly/media/javalab/backpack.png"
+            alt="backpack icon"
+            style={styles.backpackIcon}
+          />
+          <span>Backpack</span>
+        </button>
+
+        {dropdownOpen && (
```
Comment (molly-moen) [inline apps/src/javalab/Backpack.jsx:72]: does this work for both light and dark mode?

**Review by molly-moen — COMMENTED** [review]
Looks good, a couple nits/questions

File: apps/src/javalab/Backpack.jsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            ...(dropdownOpen && styles.dropdownOpenButton)
+          }}
+        >
+          <img
+            src="/blockly/media/javalab/backpack.png"
+            alt="backpack icon"
+            style={styles.backpackIcon}
+          />
+          <span>Backpack</span>
+        </button>
```
Comment (maddiedierker) [inline apps/src/javalab/Backpack.jsx]: we should use the new [JavalabButton](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/javalab/JavalabButton.jsx) component here!

File: apps/src/javalab/Backpack.jsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                </label>
+              </div>
+            ))}
+            <button
+              type="button"
+              style={styles.importButton}
+              onClick={this.collapseDropdown}
+            >
+              Import
+            </button>
```
Comment (maddiedierker) [inline apps/src/javalab/Backpack.jsx]: also use JavalabButton here

File: apps/src/javalab/Backpack.jsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    paddingRight: 8,
+    opacity: 1
+  }
+};
+
+/**
+ * A button that drops down to a set of importable files, and closes itself if
+ * you click on the import button, or outside of the dropdown.
+ */
+export const Backpack = class Backpack extends Component {
```
Comment (maddiedierker) [inline apps/src/javalab/Backpack.jsx]: nit: i think this statement is redundant -- `export class Backpack extends Component` should work the same way

File: apps/src/javalab/Backpack.jsx (hunk 4 above)
Comment (JillianK) [reply apps/src/javalab/Backpack.jsx:72]: Now it does!
Screen Shot 2021-05-27 at 1 30 24 PM

**Review by maddiedierker — APPROVED** [review]

**Review by molly-moen — APPROVED** [review]
🎉

**Pull request merged**